### ISSUE TITLE
Factor apt-get commands out

### ISSUE THREAD

**Issue by fnothaft** [BD2KGenomics/cgl-docker-lib#243]
The way we do `apt-get update && apt-get install` across many containers is kind of wonky and leads to us generating lots of layers that overlap substantially in content but that can't be cached and reused. @cket @jvivian do you all remember why we wrote all the Dockerfiles this way? Refactoring the `apt-get` update/install process should save a pretty substantial amount of build time while also reducing image size.

**Comment by cket** [comment]
No, I don't remember. Factoring out a common layer sounds good to me though!

**Comment by fnothaft** [comment]
Excellent! I'll take a hack at that sometime after #211 merges.
